sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5408,6 +5408,11 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   // reference count on the kernel, using the kernel saved in CommandData.
   PI_CALL(piKernelRetain(Kernel));
 
+  auto res=Queue->Device->getSubmitTime(*Event);
```

Comment:
can this be moved to executeCommandList?

Reply:
I would need to add a pi_event parameter to `executeCommandList` and `executeOpenCommandList`. Is that okay?

Reply:
Why don't you just record ALL events in the command-list? Maybe we should actually do this at batch submission?

Reply:
> Why don't you just record ALL events in the command-list?

Wouldn't it be redundant to record all events as their profiling information wouldn't be visible to the user?

> Maybe we should actually do this at batch submission?

I might be wrong but doesn't `executeCommandList` submit to the command to a batch if were not using immediatecommandlist ?

Reply:
The `executeCommandList` does add commands to a batch, but closes it and submits if the batch is full. This is the time when I think you should record the "command_submit" time. All events in the command-list are from these enqueue interfaces, so you are already recording submit time for all of the commands.

Reply:
Well according to SYCL [specifications](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#fig:host-acc:~:text=This%20is%20always%20some%20time%20after%20the%20command%20group%20function%20object%20returns%20and%20before%20the%20associated%20call%20to%20queue%3A%3Asubmit%20returns.), the submission time has to be calculated before the submit method returns. If we record the submission time when the batch is submitted, can't we run into a possibility that the time is recorded after `queue.submit()` returns ?

Reply:
I think you are right here, still you'd do it inside `executeCommandList`